modules/SofaGeneralEngine/src/SofaGeneralEngine/NearestPointROI.inl

```diff
@@ -75,11 +110,39 @@ void NearestPointROI<DataTypes>::reinit()
         msg_error() << "2 valid mechanicalobjects are required.";
         return;
     }
+
+    this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Valid);
     doUpdate();
 }
 
 template <class DataTypes>
 void NearestPointROI<DataTypes>::doUpdate()
+{
+    if (d_useRestPosition.getValue())
+    {
+        const VecCoord& x1 = this->mstate1->read(core::ConstVecCoordId::restPosition())->getValue();
+        const VecCoord& x2 = this->mstate2->read(core::ConstVecCoordId::restPosition())->getValue();
+
+        if (x1.size() == 0 || x2.size() == 0)
+            return;
+
+        computeNearestPointMaps(x1, x2);
+    }
+    else
+    {
+        const VecCoord& x1 = this->mstate1->read(core::ConstVecCoordId::position())->getValue();
+        const VecCoord& x2 = this->mstate2->read(core::ConstVecCoordId::position())->getValue();
+
+        if (x1.size() == 0 || x2.size() == 0)
+            return;
+
+        computeNearestPointMaps(x1, x2);
+    }
```

Comment:
I suggest to factorize this code by introducing a variable which should be equal to `position()` or `restPosition()` depending on the value of `d_useRestPosition`.

Reply:
I'm not sure to get it. I'm already factorizing using the computeNearestPointMaps(x1, x2);
I could move the 
```
if (x1.size() == 0 || x2.size() == 0)
     return;
```
at the begining of the method indeed.

Reply:
What I suggest:
```
const auto positionId = d_useRestPosition.getValue() ? core::ConstVecCoordId::restPosition() : core::ConstVecCoordId::position();
const VecCoord& x1 = this->mstate1->read(positionId)->getValue();
const VecCoord& x2 = this->mstate2->read(positionId)->getValue();

if (x1.size() == 0 || x2.size() == 0)
    return;

computeNearestPointMaps(x1, x2);
```
I reduce the amount of duplicated code by merging lines 123-129 and 133-139.

Would that work?

Reply:
yes probably, I'll test 👍